Args.cpp

```diff
@@ -107,6 +107,7 @@ void Args::parse(string line) {
     else if (key == "-cpu") { cpu = s; }
     else if (key == "-time") { timeKernels = true; }
     else if (key == "-device" || key == "-d") { device = stoi(s); }
+    if (cpu.empty()) { cpu = getHwName(device) + "-" + getBoardName(device) + "-" + str(device); }
```

Comment:
here if the user does not specify -d the variable 'device' will not contain the gpu number, to avoid this I am proposing this change:

Reply:
else if (key == "-device" || key == "-d") { device = stoi(s);
    if (cpu.empty()) { cpu = getHwName(device) + "-" + getBoardName(device) + "-" + str(device); }
}

This way only if -d is specified we assign cpu name.